packages/ws/src/ws/WebSocketShard.ts

```diff
@@ -127,6 +125,14 @@ export class WebSocketShard extends AsyncEventEmitter<WebSocketShardEventsMap> {
 	}
 
 	public async connect() {
+		const promise = this.initialConnectResolved ? Promise.resolve() : once(this, WebSocketShardEvents.Ready);
+		void this.internalConnect();
+
+		await promise;
+		this.initialConnectResolved = true;
```

Comment:
This is never reset

Reply:
Why would it? We only need it to be `false` *once*, during the initial connect. A shard that is completely `destroy`ed is cleared/re-spawned either way if re-connection is requested later for the sake of keeping things simple.